packages/frontend/src/components/textEditor/textEditor.tsx

```diff
@@ -0,0 +1,201 @@
+                '**' +
+                content.substring(start, end) +
+                '**' +
+                content.substring(end)
+        } else if (style === TextStyle.Strike) {
+            newContent =
+                content.substring(0, start) +
+                '~~' +
+                content.substring(start, end) +
+                '~~' +
+                content.substring(end)
+        } else if (style === TextStyle.Code) {
+            newContent =
+                content.substring(0, start) +
+                '`' +
+                content.substring(start, end) +
+                '`' +
+                content.substring(end)
+        } else if (style === TextStyle.Quote) {
+            newContent =
+                content.substring(0, start) +
+                '> ' +
+                content.substring(start, end) +
+                '\n' +
+                content.substring(end)
+        } else if (style === TextStyle.List) {
+            newContent = addListItem('', content, start, end)
+        }
+        setContent(newContent)
+    }
+
+    const insertImage = () => {
+        const urlComponent = document.getElementById(
+            'insert-image-url'
+        ) as HTMLInputElement
+        const url = urlComponent.value
+        if (content) {
+            const sel = document.getElementById(
+                'myTextArea'
+            ) as HTMLTextAreaElement
+            const cursor = sel.selectionStart
+            const newContent =
+                content.substring(0, cursor) +
+                `![](${url})` +
+                content.substring(cursor)
+            setContent(newContent)
+            urlComponent.value = ''
+        }
+    }
+
+    return (
+        <div>
+            <div>
+                <div className="buttons">
+                    <button onClick={() => addStyle(TextStyle.Bold)}>B</button>
+                    <button onClick={() => addStyle(TextStyle.Italic)}>
+                        I
+                    </button>
+                    <button onClick={() => addStyle(TextStyle.Strike)}>
+                        S
+                    </button>
+                    <button onClick={() => addStyle(TextStyle.Code)}>
+                        Code
+                    </button>
+                    <button onClick={() => addStyle(TextStyle.Quote)}>
+                        Quote
+                    </button>
+                    <button onClick={() => addStyle(TextStyle.List)}>
+                        List
+                    </button>
+                    <label className="insert-image">
+                        <input
+                            id="insert-image-url"
+                            type="text"
+                            placeholder="paste your image url here"
+                        />
+                        <button onClick={insertImage}>Insert Image</button>
+                    </label>
+                </div>
+                {autoFocus ? (
```

Comment:
I think this breaks because the text area id is still `myTextArea` below.

Instead of conditionally rendering here can you do this?

```
<textarea
  id="inputTextArea"
  onChange={handleContentInput}
  value={content ?? ''}
  autoFocus={autoFocus}
/>
```